spyder/widgets/sourcecode/base.py

```diff
@@ -236,7 +236,7 @@ def __init__(self, parent=None):
         BaseEditMixin.__init__(self)
         self.setAttribute(Qt.WA_DeleteOnClose)
         
-        self.extra_selections_dict = OrderedDict()
+        self.extra_selections_dict = {}

```

Comment:
Why the change is not needed? because of the `draworder` parameter?

Reply:
Partially yes, extr selections are now ordered  by the `DecorationManager`,

I moved this logic to `DecorationManager._order_decorations` this will allow that all decorations added to the editor will be order (according draw_order and selection length)